### DISCUSSION TITLE
Adding images to your doc posts

### DISCUSSION THREAD

**Discussion by james-s-w-clark** [gethyas/hyas#207] [Q&A] [answered]
I've tried having an image
1.  with my docs/doc.md file, 
2. in the images folder
3. in a new static/images folder

My image is refered to in the doc post with  `![](/images/tn.png)`
And they all get this when doing `npm run start`

```
hugo v0.88.1-5BC54738+extended windows/amd64 BuildDate=2021-09-04T09:39:19Z VendorInfo=gohugoio
Error: Error building site: "C:\Users\middo\github\myrepo\node_modules\@hyas\doks\layouts\_default\_markup\render-image.html:2:18": execute of template failed: template: _default/_markup/render-image.html:2:18: executing "_default/_markup/render-image.html" at <$image.Resize>: nil pointer evaluating resource.Resource.Resize
```

I've also tried the advice in this blog: https://snowgoons.ro/posts/2020-06-04-hugo-automated-image-processing/, so I have:

- my-doc-folder
---- _index.md
---- mydoc.md
---- awesome.png

I still get the same nil pointer with this.

______

It seems you can't use `_index.md` (list), and require `index.md` (page) to scan image files properly - https://laurakalbag.com/processing-responsive-images-with-hugo/

**Comment by james-s-w-clark** [comment, marked as answer]
[https://discourse.gohugo.io/t/how-to-access-resources-from-pages-within-a-branch-bundle/28772](https://discourse.gohugo.io/t/how-to-access-resources-from-pages-within-a-branch-bundle/28772)

> branch-bundle/branch-page.md cannot have page resources because it is not a page bundle. It cannot see branch-bundle/image.png because this image page resource belongs to branch-bundle/_index.md.
> 

so you have 

branch-bundle
......._index.md
.......page-leaf-bundle
....... ....... index.md
....... ....... image.png

and the page-leaf-bundle `index.md` can pick up the image.png!

**Comment by h-enk** [comment]
Yes that's correct.

Interesting related article: [Different ways to get an image as page resource in Hugo](https://www.markusantonwolf.com/blog/guide-for-different-ways-to-access-your-image-resources/)
